shiny/workflowPlot/server_files/model_data_plots_server.R

```diff
@@ -0,0 +1,193 @@
+# Renders ggplotly 
+
+output$modelDataPlot <- renderPlotly({
+  validate(
+    need(length(input$all_workflow_id) == 1, "Select only ONE workflow ID"),
+    need(length(input$all_run_id) == 1, "Select only ONE run ID"),
+    need(input$load_model > 0, 'Select Load Data'),
+    need(length(input$all_site_id) == 1, 'Select only ONE Site ID'),
+    need(length(input$all_input_id) == 1, 'Select only ONE Input ID'),
+    need(input$load_data > 0, 'Select Load External Data')
+  )
+  plt <- ggplot(data.frame(x = 0, y = 0), aes(x,y)) + 
+    annotate("text", x = 0, y = 0, label = "You are ready to plot!", 
+             size = 10, color = "grey")
+})
+
+# Update units every time a variable is selected
+observeEvent(input$var_name_modeldata, {
+  model.df <- load.model()
+  default.unit <- model.df %>% filter(var_name == input$var_name_modeldata) %>% pull(ylab) %>% unique()
+  updateTextInput(session, "units_modeldata", value = default.unit)
+})
+
+# Check that new units are parsible and can be used for conversion
+observeEvent(input$units_modeldata,{
+  parseable <- ifelse(udunits2::ud.is.parseable(input$units_modeldata), "can", "cannot")
+  if(udunits2::ud.is.parseable(input$units_modeldata)){
+    model.df <- load.model()
+    default.unit <- model.df %>% filter(var_name == input$var_name_modeldata) %>% pull(ylab) %>% unique()
+    if(udunits2::ud.are.convertible(default.unit, input$units_modeldata)){
+      output$unit_text2 <-  renderText({"Conversion possible"})
+    }else{
+      output$unit_text2 <-  renderText({"Units are parsible but conversion is not possible"})
+    }
+  }else{
+    output$unit_text2 <-  renderText({"Units are not parsible, please type units in udunits2 compatible format"})
+  }
+})
+
+
+
+observeEvent(input$ex_plot_modeldata,{
+  output$modelDataPlot <- renderPlotly({
+    input$ex_plot_modeldata
+    isolate({
+      masterDF <- load.model()
+      df <- dplyr::filter(masterDF, var_name == input$var_name_model)
+      updateSliderInput(session,"smooth_n_modeldata", min = 0, max = nrow(df))
+      
+      title <- unique(df$title)
+      xlab <- unique(df$xlab)
+      ylab <- unique(df$ylab)
+      
+      externalData <- load.model.data()
+
+      var = input$var_name_modeldata
+      df = df %>% select(posix = dates, vals)
+      colnames(df)[which(colnames(df) == "vals")] <- var
+      aligned_data = PEcAn.benchmark::align_data(model.calc = df, obvs.calc = externalData, var =var, align_method = "mean_over_larger_timestep")
+      colnames(aligned_data)[grep("[.]m", colnames(aligned_data))] <- "model"
+      colnames(aligned_data)[grep("[.]o", colnames(aligned_data))] <- "observations"
+      colnames(aligned_data)[which(colnames(aligned_data) == "posix")] <- "Date"
+      
+      print(head(aligned_data))
+      # Melt dataframe to plot two types of columns together
+      aligned_data <- reshape2::melt(aligned_data, "Date")
```

Comment:
No need to change this, but FWIW, the `tidyverse` analogue to `reshape2::melt` is `tidyr::gather`.